retail/interactive-tutorial/RetailSearch.Samples/SearchWithFacetSpecSample.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START retail_search_product_with_facet_spec]
+
+using Google.Api.Gax;
+using Google.Cloud.Retail.V2;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+/// <summary>
+/// Search with facet spec sample class.
+/// </summary>
+public class SearchWithFacetSpecSample
+{
+    /// <summary>Get search request.</summary>
+    private SearchRequest GetSearchRequest(string query, string facetKeyParam, string projectNumber)
+    {
+        string defaultSearchPlacement = $"projects/{projectNumber}/locations/global/catalogs/default_catalog/placements/default_search";
+
+        var facetKey = new SearchRequest.Types.FacetSpec.Types.FacetKey
+        {
+            Key = facetKeyParam
+        };
+
+        // Put the intervals here:
```

Comment:
@amanda-tarafa 
I apologize for the confusion regarding the "intervals" comment. Actually, it is needed for the testing of specific scenario. You can find the desription of scenario of adding the intervals to the sample in Tutorial file for SearchWithFacetSpec Sample (lines 202-224) :
https://github.com/t-karasova/dotnet-docs-samples/pull/2/files#diff-9c0c6ef57d930c4614ee913048bb130b8a4c1a00dfa527011acea4da4ae4c801

Reply:
In case of this specific scenario we have to declare the **facetSpec** variable separately, in order to add the Intervals to it.